### PR TITLE
Add read group back and crumble for HiFi

### PR DESCRIPTION
**Pull request by priyanka-surana**

### CODE REVIEW

**Review by muffato — COMMENTED** [review]
Is there a small PacBio file we could use in the test profile ? Had to run the _full_ test profile to see the outcome of this change.

Shane also suggested to add the `-c` and `-p` options to `samtools merge` in order to remove redundant `@RG` (and `@PG`) headers. Could you add that ?

File: conf/modules.config [hunk 1]
```diff
@@ -81,6 +85,16 @@ process {
         ext.prefix = { "${input.baseName}" }
     }
 
+    withName: CRUMBLE {
+        ext.prefix = { "${input.baseName}.crumble" }
+        ext.args   = '-y pbccs -O cram'
+        publishDir = [
+            path: { "${params.outdir}/read_mapping/crumble" },
```
Comment (muffato) [inline conf/modules.config:92]: The crumbled file is meant to replace the uncompressed cram file, so it should be in the same directory
```suggestion
            path: { "${params.outdir}/read_mapping/pacbio" },
```

File: modules/nf-core/crumble/main.nf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    script:
+    def args       = task.ext.args ?: ''
+    def prefix     = task.ext.prefix ?: "${meta.id}"
+    def extension  = args.contains("-O sam") ? "sam" :
+                    args.contains("-O bam") ? "bam" :
+                    args.contains("-O cram") ? "cram" :
+                    "sam"
+    def bedin      = keepbed ? "-R ${keepbed}" : ""
+    def bedout     = bedout ? "-b ${prefix}.out.bed" : ""
```
Comment (muffato) [inline modules/nf-core/crumble/main.nf:33]: I would suggest renaming this BED file to make the name more informative. The help page says
```
-b out.bed        Output suspicious regions to out.bed []
```
So maybe `${prefix}.suspicious_regions.bed` ?

File: conf/modules.config (hunk 1 above)
Comment (priyanka-surana) [reply conf/modules.config:92]: For now we are publishing both CRAM and crumble CRAM. Once we add crumble feature for the other technologies and add the output options, I will update this. Right now I don't know what that will look like.

Comment (muffato) [reply conf/modules.config:92]: The requirement is to have the directory called `read_mapping/${library_type}`. The names of the crumbled files don't clash with the pre-crumble files, so all could co-exist within `pacbio/`

File: modules/nf-core/crumble/main.nf (hunk 2 above)
Comment (priyanka-surana) [reply modules/nf-core/crumble/main.nf]: Done ✅

File: conf/modules.config (hunk 1 above)
Comment (priyanka-surana) [reply conf/modules.config]: Done ✅

**Comment by priyanka-surana** [comment]
I updated the `docker.registry` which led to updating lots of modules.

**Review by muffato — COMMENTED** [review]
My "full_test" run stopped because of a RUNLIMIT / exit code 140 on `SANGERTOL_READMAPPING:READMAPPING:ALIGN_ONT:MINIMAP2_ALIGN (ilPolIcar4_T1)`.
`errorStrategy` in `conf/base.config` indeed doesn't contain the 140. Could you add that ? or make it match the nf-core template ? (which now has a range)

**Comment by priyanka-surana** [comment]
> My "full_test" run stopped because of a RUNLIMIT / exit code 140 on `SANGERTOL_READMAPPING:READMAPPING:ALIGN_ONT:MINIMAP2_ALIGN (ilPolIcar4_T1)`.
> `errorStrategy` in `conf/base.config` indeed doesn't contain the 140. Could you add that ? or make it match the nf-core template ? (which now has a range)

This might be because of the change in `sanger.config` because with the old version we had a default retry strategy. I have made the update, but will push later, since I was in the middle of some other changes too.

**Review by muffato — APPROVED** [review]
Approving as I have no concerns about the code, and it's been used in production for a few months !

**Pull request merged**